float32 x
float32 y
--------

float32 x_new
float32 y_new